gh-119609: Add PyUnicode_Export() and PyUnicode_Import() functions (#119610)
python/cpython: Doc/c-api/unicode.rst

   .. versionadded:: 3.3

.. c:function:: const void* PyUnicode_Export(PyObject *unicode, uint32_t requested_formats, Py_ssize_t *nbytes, uint32_t *format)

   Export the contents of the *unicode* string in one of the requested format
   *requested_formats*.

   * On success, set *\*nbytes* and *\*format*, and return the contents.
   * On error, set an exception and return ``NULL``.

>    The contents is valid as long as *unicode* is valid.
> 
>    The export must be released by :c:func:`PyUnicode_ReleaseExport`.

Reviewer: ```suggestion
   The export must be released by :c:func:`PyUnicode_ReleaseExport`.
   The contents of the buffer are valid until they are released.

   The buffer must not be modified.

   To help avoid buffer overflows, the exported buffer is followed
   by a zero codepoint. This terminator is not counted in *nbytes*. 
```

   *unicode*, *nbytes* and *format* must not be NULL.

   Available formats:

   .. c:namespace:: NULL

   ===================================  ========  ===========================

>    Constant Identifier                  Value  Description

Reviewer: ```suggestion
   Constant Identifier                    Value  Description
```

   ===================================  ========  ===========================
   .. c:macro:: PyUnicode_FORMAT_ASCII  ``0x01``  ASCII string (``Py_UCS1*``)
   .. c:macro:: PyUnicode_FORMAT_UCS1   ``0x02``  UCS-1 string (``Py_UCS1*``)
   .. c:macro:: PyUnicode_FORMAT_UCS2   ``0x04``  UCS-2 string (``Py_UCS2*``)
   .. c:macro:: PyUnicode_FORMAT_UCS4   ``0x08``  UCS-4 string (``Py_UCS4*``)
   .. c:macro:: PyUnicode_FORMAT_UTF8   ``0x10``  UTF-8 string (``char*``)
   ===================================  ========  ===========================

   *requested_formats* can be a single format or a combination of the formats

>    in the table above.

Reviewer: ```suggestion
   *requested_formats* can be a single format or a bitwise combination of the
   formats in the table above.
   On success, *\*format* will be set to a single one of the requested flags.
```

   Note that future versions of Python may introduce additional formats.

   .. versionadded:: 3.14

.. c:function:: void PyUnicode_ReleaseExport(PyObject *unicode, const void* data, uint32_t format)

>    Release an export created by :c:func:`PyUnicode_Export`.

Reviewer: ```suggestion
   Release an export created by :c:func:`PyUnicode_Export`.

   Each argument must match the corresponding argument or result of
   a single earlier call to :c:func:`PyUnicode_Export`.
   In particular, this means that you must hold a reference to *unicode*
   while an export is valid.
```
